Convert component scss into mixins + import all styles into index.scss file.

Caleb-Ellis opened the pull request:
## Done
- Converted all component scss into `_index.scss`'s (for nicer import paths) and removed them from being directly imported into React components.
- Wrapped all "global" styles in mixins.
- Created a top-level `index.scss` file where all scss in the project is listed. I think this is nicer because now we only have to import vanilla+settings only once, instead of as needed per-component (which I believe was also causing some settings overrides).
- Upgraded to Vanilla 2.8.0 and fixed regressions + removed overrides (mainly to do with contextual menu positioning and form spacing).
- Removed some "easy win" redundant styles that were either fixed in the latest Vanilla or copied unnecessarily from legacy, but we should definitely go over what we have more thoroughly.

## QA
- Click around everywhere in the React pages and check that styling still looks ok everywhere.
- See if you agree with the folder structure

## Fixes
Fixes #747

huwshimi commented:
Thanks for doing all these changes. I've noticed a couple of things, it's possible that some are on master but worth fixing (could be in a followup if needed).

Generally the forms that are in a card are missing space between the last input and the horizontal line:
<img width="721" alt="Screen Shot 2020-03-11 at 10 45 37 am" src="https://user-images.githubusercontent.com/361637/76369161-b87a1f00-6386-11ea-892e-9a3f4b5242d2.png">

The forms that aren't in a card are generally OK but I noticed this one in the user preferences:
<img width="578" alt="Screen Shot 2020-03-11 at 10 46 31 am" src="https://user-images.githubusercontent.com/361637/76369164-bc0da600-6386-11ea-9e35-e3b40e0f5be2.png">

The DHCP snippets have a bit of extra space below them:
<img width="923" alt="Screen Shot 2020-03-11 at 10 49 59 am" src="https://user-images.githubusercontent.com/361637/76369167-bd3ed300-6386-11ea-9ad0-f5555ab05db4.png">

huwshimi commented on ui/src/app/base/components/ContextualMenu/ContextualMenu.js:
```diff
@@ -9,7 +9,11 @@ const getPositionStyle = (position, wrapper, constrainPanelWidth) => {
+    bottom,
+    left,
+    width
+  } = wrapper.current.children[0].getBoundingClientRect();
```
Instead of accessing the children would it be possible to move the ref to the correct node?

Caleb-Ellis replied:
Hmm I can't pass it to the `Button` because you can only assign `ref`s to normal DOM elements, and there isn't currently a `useForwardRef` hook. I've added a guard in the function so that if a child isn't present it just gets the dimensions of the span, but otherwise it gets the first child's dimensions. Would that be ok for now?

huwshimi replied:
Yep, that's better I reckon.

huwshimi commented on ui/src/scss/_vanilla-overrides.scss:
```diff
@@ -4,17 +4,6 @@
-// 19-08-2019 Caleb: Form submit button too close to input when $multi = 1
-// https://github.com/canonical-web-and-design/vanilla-framework/issues/2497
-.p-form__group:last-child {
-  margin-bottom: $spv-outer--medium;
```
As noted in the QA notes maybe this one can't be removed yet?

Caleb-Ellis replied:
Upgrading to Vanilla 2.8.0 meant this rule broke a few other things. :( I've left it removed but I've added another issue for a slightly different regression.

Caleb-Ellis commented:
> Generally the forms that are in a card are missing space between the last input and the horizontal line:
> <img alt="Screen Shot 2020-03-11 at 10 45 37 am" width="721" src="https://user-images.githubusercontent.com/361637/76369161-b87a1f00-6386-11ea-892e-9a3f4b5242d2.png">

@huwshimi I think this looks fine given that the spacing is supposed to be dense. Having collated the custom scss we have in the project (and seen how much it is) I'd rather let Vanilla define the spacing, and only have an override if it looks truly awful.



> The forms that aren't in a card are generally OK but I noticed this one in the user preferences:

Fixed by adding a little spacer class to the div. I think that's fine given this is a bit of a special case i.e. it's not an input in the form.

![screenshot_8](https://user-images.githubusercontent.com/25733845/76407827-3e4c9900-638c-11ea-8dd4-ae30b5d8e886.png)



> The DHCP snippets have a bit of extra space below them:

Thanks. Fixed.

![screenshot_9](https://user-images.githubusercontent.com/25733845/76407867-53292c80-638c-11ea-832f-672ebd16f2ef.png)

huwshimi commented:
> > Generally the forms that are in a card are missing space between the last input and the horizontal line:
> > <img alt="Screen Shot 2020-03-11 at 10 45 37 am" width="721" src="https://user-images.githubusercontent.com/361637/76369161-b87a1f00-6386-11ea-892e-9a3f4b5242d2.png">
> 
> @huwshimi I think this looks fine given that the spacing is supposed to be dense. Having collated the custom scss we have in the project (and seen how much it is) I'd rather let Vanilla define the spacing, and only have an override if it looks truly awful.

That's fine, it just differs from the designs, but maybe check with them if you're concerned.
